### PR TITLE
Refactor `test_intermediate_plot.py`

### PR DESCRIPTION
**Pull request by contramundum53**
## Motivation
This is a followup of #3731. We refactor the tests of `plot_intermediate_values`.

## Description of the changes
Refactor the tests of `plot_intermediate_values`.

We first test whether `_IntermediatePlotInfo` is correct. Then, we test whether `_intermediatePlotInfo` is plotted correctly.

### CODE REVIEW

**Comment by toshihikoyanase** [comment]
@nzw0301 @keisuke-umezawa Could you review this PR, please? If you are busy, please feel free to remove the assignment.

**Comment by HideakiImamura** [comment]
@contramundum53 Could you resolve conflicts?

File: tests/visualization_tests/test_intermediate_plot.py [hunk 1]
```diff
@@ -1,53 +1,63 @@
 from optuna.study import create_study
```
[the remark this answers is not in the document]
Comment (HideakiImamura) [reply tests/visualization_tests/test_intermediate_plot.py:7]: @contramundum53 What do you think?

**Review by HideakiImamura — APPROVED** [review]
Thanks, LGTM! But, could you answer the question from @keisuke-umezawa?

File: tests/visualization_tests/test_intermediate_plot.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (contramundum53) [reply tests/visualization_tests/test_intermediate_plot.py:7]: Sorry, I forgot to reply to this.
I agree that this increases readability and I fixed it. Thanks.

**Comment by c-bata** [comment]
Let me assign me as a reviewer since umezawa-san is busy these days.

**Review by c-bata — APPROVED** [review]
Changes looks great to me!

**Pull request merged**